lib/sales_analyst.rb

```diff
@@ -80,11 +86,129 @@ def standard_deviation(values, mean)
     std_dev.round(2)
   end
 
+  def average_invoices_per_merchant
+    average = all_invoices.length.to_f / all_merchants.length
+    average.round(2)
+  end
+
+  def num_of_invoices_per_merchant
+    all_merchants.each_with_object({}) do |merchant, total_per_merchant|
+      total_invoices = all_invoices.count do |invoice|
+        invoice.merchant_id == merchant.id
+      end
+      total_per_merchant[merchant] = total_invoices
+    end
+  end
+
+  def average_invoices_per_merchant_standard_deviation
```

Comment:
All looks great Noah! 🙌🏼 🎉 Am wondering if we can refactor this method name later to be a bit shorter?

Reply:
Unfortunately that's from the spec so it has to stay this way